facia-press/app/controllers/Application.scala

```diff
@@ -20,4 +24,21 @@ object Application extends Controller with ExecutionContexts {
     FrontPress.pressLiveByPathId(path).map(Json.stringify(_))
       .map(Ok.apply(_))
       .map(NoCache.apply)}
+
+  private def handlePressRequest(path: String, liveOrDraft: String)(f: (String) => Future[_]): Future[Result] =
+    if (FaciaPressOnDemand.isSwitchedOn) {
+      val stage = Configuration.facia.stage.toUpperCase
+      f(path)
```

Comment:
What is `f()` - is it just the same as `Future.successful`?

If so then just make these non async.